### DISCUSSION TITLE
issue: Rendering a controller for a field in field array causes issues with setting value for that field

### DISCUSSION THREAD

**Discussion by janKozie1** [react-hook-form/react-hook-form#6799] [Q&A]
### Version Number

7.17.2

### Codesandbox/Expo snack

https://codesandbox.io/s/react-hook-form-js-forked-b8kps?file=/src/App.js

### Steps to reproduce

1. Go to the codesandbox link
2. Open console
3. Click on the `change all values to 6` button
4. Note how the value for a field which has an associated controller resets back to the original value

### Expected behaviour

Whether there is or not a controller associated with a field in field array, should not have impact on the ability to `setValue` of that field.

### What browsers are you seeing the problem on?

Firefox, Chrome, Safari

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
if you use it properly, it's working as expected: https://codesandbox.io/s/react-hook-form-js-forked-r1kli?file=/src/App.js

**Comment by janKozie1** [comment]
Thank you for the quick response! 
Is there a way to make the ids persists? I use them to determine if the user added new fields (by comparing them to response from my API).

**Comment by bluebill1049** [comment]
change `keyName` prop to something else, so `id` can be your own formValues.
